test/test_torch.py

```diff
@@ -5165,12 +5171,14 @@ def _test_memory_format_transformations(self, device, input_generator_fn, transf
         if compare_data:
             self.assertEqual(xc, clone.to(xc))
 
-        x = torch.randn((3, 4, 5, 6, 7, 8, 9), device=device)
-        for _ in range(10):
-            permutation = list(range(len(x.shape)))
-            random.shuffle(permutation)
-            x = x.permute(permutation)
-            self.assertEqual(x.stride(), transformation_fn(x, memory_format=torch.preserve_format).stride())
+        # TODO copy _like constructors to stride permutation instead of layout
```

Comment:
From your commit message, it sounds like the TODO is to have `_like` constructors preserve permutations as well as the layout. Is that right? I don't understand why you say "instead of" here

Reply:
That's correct. meant to be `instead of just layout`